bcgov/HOUS-permit-portal
# Feature/bphh 1797/permit collaborations collaborator permit application viewing, updating with given permissions

## Description
- This PR includes:
   - Allowing collaborators to view permit applications they have access to from index page with indicator
   - Allows them to update (save) draft permit applications based on their edit permissions. Designated submitter: can submit ad edit all blocks, assignees can only edit blocks they are assigned to
   - UI elements to block/ indicate what users can edit. They should be only able to edit and save the blocks they are assigned to or for designated submitters they can edit all blocks and submit. They cannot edit any other permit application details


## What type of PR is this? (check all applicable)

- [x ] 🍕 Feature

## Related Tickets & Documents
https://hous-bssb.atlassian.net/browse/BPHH-1797


## Steps to QA
- Pre req: follow the QA steps in this PR https://github.com/bcgov/HOUS-permit-portal/pull/599 to assign a collaborator to a permit application
- Log in as assigned collaborator
- Should be able to view permit application in the list
![Screenshot 2024-07-31 at 10 28 04 AM](https://github.com/user-attachments/assets/88ebceb5-82dd-4683-92ec-4cc97069128e)
- When entering the permit application. They should only have access and be able to update the requirement blocks they are assigned to
- Original Permit Application:
![Screenshot 2024-07-31 at 10 36 31 AM](https://github.com/user-attachments/assets/33d880a3-4c39-4cf3-9d73-60d49b96dfef)
- Permit Application from collaborator view
![Screenshot 2024-07-31 at 10 28 16 AM](https://github.com/user-attachments/assets/b715dd67-82ba-46f1-a30e-7d04a73edd31)
- For collaborators who are not designated submitters. They should get the following modal when trying to submit
![Screenshot 2024-07-31 at 10 28 23 AM](https://github.com/user-attachments/assets/e2b089fe-97cb-4581-9ba0-4a9ffc337b5b)
- Collaborators shouldn't be able to edit any other permit details such as nickname etc
